syft/frameworks/torch/tensors/interpreters/autograd.py

```diff
@@ -338,11 +338,13 @@ def detail(worker: AbstractWorker, tensor_tuple: tuple) -> "AutogradTensor":
             id=syft.serde.msgpack.serde._detail(worker, tensor_id),
             requires_grad=requires_grad,  # ADDED!
             preinitialize_grad=preinitialize_grad,
-            grad_fn=grad_fn,
             # local_autograd=local_autograd,
             data=chain,  # pass the de-serialized data
             tags=syft.serde.msgpack.serde._detail(worker, tags),
             description=syft.serde.msgpack.serde._detail(worker, description),
         )
 
+        # Assigning grad_fn here. grad_fn doesn't get assigned when passed in constructor
+        tensor.grad_fn = syft.serde.msgpack.serde._detail(worker, grad_fn)
```

Comment:
Would you be willing to fix the `AutogradTensor.__init__`  to fix this? Should be very quick

Reply:
Fixed.